### PR TITLE
Added nullability qualifiers to YMAPIClient

### PR DESCRIPTION
**Pull request by pwillsey**
Added the new nullability qualifiers to the API Client to help interoperability with Swift.

### CODE REVIEW

File: OAuthSDK/YMAPIClient.m [hunk 1]
```diff
@@ -108,9 +105,9 @@ - (NSString *)userAgent
 }
 
 - (void)getPath:(NSString *)path
```
Comment (joelklabo) [inline OAuthSDK/YMAPIClient.m]: should the path parameter be 'nonnull'?

Comment (joelklabo) [inline OAuthSDK/YMAPIClient.m]: Oh, I see you did the assume nonnull thing. Never mind.

**Comment by joelklabo** [comment]
Looks good.

File: OAuthSDK/YMAPIClient.h [hunk 2]
```diff
@@ -23,20 +23,27 @@
...
-@property (nonatomic, copy) NSString *authToken;
+
+@property (nonatomic, copy, nullable) NSString *authToken;
 
 /**
  Default initializer.
  @param authToken The OAuth token.
  */
-- (id)initWithAuthToken:(NSString *)authToken;
+- (instancetype)initWithAuthToken:(nullable NSString *)authToken;
```
Comment (MarcoSero) [inline OAuthSDK/YMAPIClient.h]: I think we should be consistent and use the order Apple's using which I think is `type nullability_annotation`.

**Comment by MarcoSero** [comment]
:ship: it!

File: OAuthSDK/YMAPIClient.h (hunk 2 above)
Comment (pwillsey) [inline OAuthSDK/YMAPIClient.h]: I got the impression Apple wants you to use the non-underscored version where possible:

_You can use __nullable and __nonnull almost anywhere you can use the normal C const keyword, though of course they have to apply to a pointer type. However, in the common case there’s a much nicer way to write these annotations: within method declarations you can use the non-underscored forms nullable and nonnull immediately after an open parenthesis, as long as the type is a simple object or block pointer._

This came from: https://developer.apple.com/swift/blog/?id=25

Comment (MarcoSero) [inline OAuthSDK/YMAPIClient.h]: I just meant the order of the keywords :)
In this case
```objective-c
- (instancetype)initWithAuthToken:(NSString * nullable)authToken;
```

Comment (pwillsey) [inline OAuthSDK/YMAPIClient.h]: You can only use the non underscored version after an open parenthesis, otherwise it's a syntax error I think

Comment (jasonhotsauce) [inline OAuthSDK/YMAPIClient.h]: You can use `- (instancetype)initWithAuthToken:(nullable NSString *)authToken;` or `- (instancetype)initWithAuthToken:(NSString *__nullable)authToken;`

**Pull request merged**